test/livebook_web/live/session_live_test.exs

```diff
@@ -1171,6 +1171,199 @@ defmodule LivebookWeb.SessionLiveTest do
              |> has_element?()
     end
 
+    test "configuring k8s runtime", %{conn: conn, session: session} do
+      {:ok, view, _} = live(conn, ~p"/sessions/#{session.id}/settings/runtime")
+
+      Session.subscribe(session.id)
+
+      Req.Test.stub(Livebook.Runtime.K8s, Livebook.ClusterStub)
+      Req.Test.stub(Livebook.Runtime.K8s.NoPermissionCluster, Livebook.NoPermissionClusterStub)
```

Comment:
`Livebook.Runtime.K8s` here has nothing to do with the actual module, so I would change the naming for clarity:

```suggestion
      Req.Test.stub(:k8s_cluster, Livebook.K8sClusterStub)
      Req.Test.stub(:k8s_no_permission_cluster, Livebook.K8sNoPermissionClusterStub)
```

Sidenote: just checking, there is no way to detect the context from the request?

Reply:
Not sure I get what you mean... I'd like to test the context switcher. Therefore I need to have 2 contexts in my Kubeconfig (see test.exs and `Kubereq.Kubeconfig.Stub`).

Reply:
I was wondering if we can have a single plug implementation that we would stub for both contexts. But this would only work if we can get context from request, so we can deny permission if the request comes for "no-permission".

Reply:
yeah I see. gotta think about this. :)

I guess if `Kubereq.Kubeconfig.Plug` would set the server_url to something like `"https://#{context_name}"` [here](https://github.com/mruoss/kubereq/blob/v0.1.7/lib/kubereq/kubeconfig/stub.ex#L123), we could decide based on that. But I'd have to patch `kubereq` first.

Reply:
Yup, this worked. What do you think of it now?

Reply:
Looks good!